core/module_specifier.rs

```diff
@@ -1,6 +1,30 @@
 use std::fmt;
```

Comment:
I believe this module deservers it's own error kind. Besides wrapping `url::ParseError` we need to handle:
- unsupported prefix for Windows paths - `ensure_valid_prefix`
- situation no 2 from `ModuleSpecifier::resolve` - I don't have an idea what to name this error, but it causes a lot of trouble for people (#2338)